modules/FIC_MainMenu.py

```diff
@@ -63,63 +65,87 @@ def _set_argument_flags(self, args):
 
         return
 
-    def repo_selection_menu(self):
-        pass
+    def _choice_main_menu(self):
 
-    def main_menu(self):
-        self._set_argument_flags(self.parse_arguments())  # Set all flags before showing the menu.
-
-        print("Welcome to Ciduty's Firefox Infra Changelog!\n"
-              "You can use the options below to run the script according to your needs.\n"
-              "\n"
-              "1. Run script for all available repositories \n"
-              "2. Run script only for repositories that are on GitHub\n"
-              "3. Run script only for repositories that are on Mercurial\n"
-              "4. Run script only for repositories that are chosen by you (Both GitHub and/or Mercurial)\n"
-              "5. Activates logger output in console\n"
-              "6. Generates changelog.md for the amount of days set by user\n"
-              "7. Run the script for all available repositories and pushes .json/.md changes to Git\n"
-              "0. Exit application.")
-
-        choice = input()
-
-        if choice == 1:
+        if self.choice is 1:
             self.all = True
+            self._check_arguments_state()
 
-        if choice == 2:
+        if self.choice is 2:
             self.git_only = True
+            self._check_arguments_state()
 
-        if choice == 3:
+        if self.choice is 3:
             self.hg_only = True
+            self._check_arguments_state()
 
-        if choice == 4:
+        if self.choice is 4:
             self.repo = True
 
-        if choice == 5:
+        if self.choice is 5:
             self.logging = True
+            self.main_menu()
 
-        if choice == 6:
+        if self.choice is 6:
             days_choice = input("Please enter the number of days you want changelog.md to be generated for\n")
             if str(days_choice).isdecimal():
                 self.days = days_choice
             else:
                 print("When using option 6 please use integers only.")
                 exit(4)
 
-        else:
-            exit(0)
+        return
```

Comment:
If we can choose only one option from the menu, should we add a 'return' at each of the "if self.option is  int():" blocks of the "_choice_main_menu(self)" method.  This will stop to check the rest of the conditions once one is fulfilled.

Reply:
Basically the menu will pop up again only if the chosen option is 5 (activates logger) . If any other option is chosen, the script shall start to work on that matter.